frame/composable-traits/src/dex.rs

```diff
@@ -33,21 +33,33 @@ pub trait Amm {
 	fn lp_token(pool_id: Self::PoolId) -> Result<Self::AssetId, DispatchError>;
 
 	/// Returns the amount of base & quote asset redeemable for given amount of lp token.
-	fn redeemable_assets_for_given_lp_tokens(
+	fn redeemable_assets_for_lp_tokens(
 		pool_id: Self::PoolId,
 		lp_amount: Self::Balance,
+		min_expected_amounts: Vec<Self::Balance>,
 	) -> Result<RedeemableAssets<Self::AssetId, Self::Balance>, DispatchError>
 	where
 		Self::AssetId: sp_std::cmp::Ord;
 
 	/// Returns the amount of LP tokens that would be recieved by adding the given amounts of base
 	/// and quote.
-	fn amount_of_lp_token_for_added_liquidity(
+	fn simulate_add_liquidity(
```

Comment:
docs should be canged too. so may be in other PR not to block merge. i am very agains adding new work into already decide scope until work doen is useles. it is not useless in this case.

Reply:
updated.